contrib/eraser/training/train_eraser.py

```diff
@@ -551,85 +632,97 @@ def get_filter_mappers(resolution: int) -> list[MapperWrapper | KeyFilter]:
         ),
     ]
 
-    return filters_mappers
+@dataclass
+class ShardConfig:
+    path: str
+    type: str # "inpainter" or "eraser"
```

Comment:
nit: I would not name this `type` (due to https://docs.python.org/3/library/functions.html#type). `kind` maybe? (see also the YAML configs then).

Also consider using https://docs.python.org/3/library/typing.html#typing.Literal (for "inpainter" or "eraser")